### ISSUE TITLE
Does trials resume automatically?

### ISSUE THREAD

**Issue by keunwoochoi** [keras-team/keras-tuner#439]
Hi, thanks for the great library. I'm trying to use keras-tuner for the first time. 

My current computing environment is quite unstable so in my terminal, I let the python command that runs keras-tuner search to be re-executed once the process is terminated. Then I got a Tensorboard results like below. 

![image](https://user-images.githubusercontent.com/16153797/98625512-b8e40480-22dd-11eb-9c3d-fcd368c3470b.png)

There are only two trials, and the second one seems to be stopped and re-run again and again. (Btw I'm passing a fixed seed during instantiating the tuner.)

From the implementations of `Oracle.create_trial()`,  `BaseTuner.search()`, and `BaseTuner.on_trial_begin(trial))`, and this behavior I'm seeing, I think keras tuner does not resume if a training was stopped in the middle of a trial.
Am I correct? And what would be a fix? To implement resuming-from-the-latest-checkpoint, seems like, I have to make quite a lot of changes (inheriting the tuner classes and override several methods, etc) which would be only a temporary solution. But I'm not sure if there's any other solution.

**Comment by keunwoochoi** [comment]
AND, for the tuner, shouldn't I pass an EarlyStopping callback? I suspect if it is stopping the training after a few epochs but the tuner doesn't know about it, hence re-starting the trial, ending up at an infinite loop like you see in the tensorboard.

**Comment by keunwoochoi** [comment]
![image](https://user-images.githubusercontent.com/16153797/98856488-9bbd4c00-242b-11eb-850c-6fa28661f9a4.png)
